JustSaying.Models/Message.cs

```diff
@@ -3,6 +3,19 @@
 
 namespace JustSaying.Models
 {
+    public class PublishEnvelope
```

Comment:
Just checking, this is a _publish-only_ thing? So a subscriber would never interact with an envelope? If they do, I'd suggest `MessageEnvelope`.

Reply:
I think it might cause confusion if the envelope on the receive side is named differently from the one on the publish side. Would they have different members?

Reply:
They would likely have different members, such as "approximate receive count". So I think it makes sense to have different versions, but keep the pattern the same.

Reply:
They would have different members and/or implementation of  such, yes.